### ISSUE TITLE
New Rule: no-irregular-whitespace

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#157]
### Rule details

Disallow irregular whitespaces.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
    .firstClass .secondClass {}
/**            ↑
 * Irregular whitespace. Selector would fail to match '.firstClass' */
```

### Prior Art

[no-irregular-whitespace](https://stylelint.io/user-guide/rules/no-irregular-whitespace/)

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Various whitespace characters can be inputted by programmers by mistake for example from copying or keyboard shortcuts. Pressing Alt + Space on macOS adds in a non breaking space character for example.

**Comment by nzakas** [comment]
Thanks for the suggestion. I don't think it makes sense to implement this rule. We already have [`no-irregular-whitespace`](https://eslint.org/docs/latest/rules/no-irregular-whitespace) in ESLint, and there's probably a way to make it generic so it can be used in any language. I'd rather focus on that instead of duplicating the logic for each language.
